scene/2d/navigation_region_2d.cpp

```diff
@@ -336,7 +336,7 @@ void NavigationRegion2D::_bind_methods() {
 
 	ClassDB::bind_method(D_METHOD("_navigation_polygon_changed"), &NavigationRegion2D::_navigation_polygon_changed);
 
-	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "navigation_polygon", PROPERTY_HINT_RESOURCE_TYPE, "NavigationPolygon"), "set_navigation_polygon", "get_navigation_polygon");
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "navigation_polygon", PROPERTY_HINT_RESOURCE_TYPE, "NavigationPolygon", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_EDITOR_INSTANTIATE_OBJECT), "set_navigation_polygon", "get_navigation_polygon");
```

Comment:
The editor already has a dedicated popup dialog for the navigation region to add a resource when users interact with the polygon tools or baking. Auto-adding is imo not good because new users will easily have embedded resources in their scene files that can end up really big. The popup or manual add at least makes them aware of the resource.